sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -408,7 +408,15 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   private def defaultTablePath(tableIdent: TableIdentifier): String = {
-    val dbLocation = getDatabase(tableIdent.database.get).locationUri
```

Comment:
how about we put this logic in `ExternalCatalog`? e.g. `ExternalCatalog.getDatabase` will check the default database and then call `getDatabaseInternal` which need to be implemented by subclasses.